src/script/client/ClientEntity.ts

```diff
@@ -84,21 +81,21 @@ export class ClientEntity {
    * @returns `true`, if the client is the self user's permanent client.
    */
   isPermanent(): boolean {
-    return this.type === ClientType.PERMANENT;
+    return this.isSelfClient && this.type === ClientType.PERMANENT;
   }
 
   /**
    * @returns `true`, if it is **not** the client of the self user.
    */
   isRemote(): boolean {
-    return !this.isPermanent() && !this.isTemporary();
+    return !this.isSelfClient && !this.isPermanent() && !this.isTemporary();
   }
```

Comment:
I just checked and it seems as if this method is not used anywhere besides tests.